### PR TITLE
ci: use CodeQL instead of LGTM

### PR DESCRIPTION
**Pull request by mrc0mmand**
As LGTM is going to be shut down by EOY[0], let's move the code scanning to CodeQL as recommended. Thanks to GH integration the results from such scans will be shown both in the respective PR and in the Security -> Code Scanning tab[1].

[0] https://github.blog/2022-08-15-the-next-step-for-lgtm-com-github-code-scanning/
[1] https://github.com/dracutdevs/dracut/security/code-scanning

---

Going through my LGTM dashboard I noticed that dracut has LGTM enabled[0] but without the PR integration[1], hence I'm not sure to what degree you actually utilize the LGTM alerts. In other words - if you don't find LGTM/CodeQL useful, feel free to disregard this PR and just close it.

~~(I had to open this PR from the upstream repo itself, not my fork, otherwise GH would refuse to pick up the new action...)~~
Nevermind, that was caused by a typo, oh well... :shrug: 

[0] https://lgtm.com/projects/g/dracutdevs/dracut/
[1] https://lgtm.com/projects/g/dracutdevs/dracut/ci/

### CODE REVIEW

**Comment by mrc0mmand** [comment]
The newly-found alerts can be found at https://github.com/dracutdevs/dracut/security/code-scanning?query=pr%3A1987+tool%3ACodeQL+is%3Aopen - once (if) this is merged, they'll appear here - https://github.com/dracutdevs/dracut/security/code-scanning where they can be dismissed and won't appear again.

As for the "Code scanning results" fail - this won't happen again in other PRs unless they introduce new alerts themselves (currently the CodeQL didn't run on the base (master) branch, so all alerts are marked as new).

Also, as I enabled the extended checks by default, it might show a bit more alerts that won't be deemed useful - if that's the case, the CodeQL config can be fine-tuned by enabling/disabling specific checks or query categories. PTAL!

**Review by johannbg — APPROVED** [review]
LGTM however I do believe we need to fix all the detected issues the scanner detected prior to implementing it ( have a clean state to start from ) atleast from my pov scanners are entirely useless if the detected output from them is just silenced.

**Review by LaszloGombos — APPROVED** [review]
LGTM, as long as there is an honest attempt to get to a baseline and resolve the issues.

File: .github/workflows/codeql.yml
Comment (LaszloGombos) [inline .github/workflows/codeql.yml:32]: Is python necessary ?

File: .github/workflows/codeql.yml
Comment (LaszloGombos) [inline .github/workflows/codeql.yml:21]: All other bots use  ubuntu-latest

**Comment by johannbg** [comment]
Adding this to the 059 milestone and removing the need discussion label since this has already been acked,

**Pull request closed without merging**
